Lib/TLib.hs

```diff
@@ -242,12 +240,16 @@ failmsg :: String -> String
 failmsg s = "failure to say '{}'"%s
 
 -- | penalize if 's' is not said
-mustSay s = when (not_$ said s) (doBefore (penalty 1 (failmsg s)))
+mustSay s = mustSay' s failmsg
+mustSay' s failmsg = when (not_$ said s) (doBefore (penalty 1 (failmsg s)))
```

Comment:
Can you think of a case where the argument `failmsg` would care about `s` (which may be a regex) and you don't just want `mustSay`? In the samples, it's always `const somthing`, so adding that here might be better.
There might also be a case for abstracting away the whole penalty - think of a rule which updates some counter or insults people rather than giving them cards.

Reply:
I don't think so, since unlike with sometimesSay and unnec the function is only ever called with the argument to mustSay itself rather than with what was said.
So it's probably best to make this argument a String

Reply:
I don't think the penalty needs to be abstracted away - the use case is narrow enough that `when (not_$ said s)` probably suffices and is more clear to read